Gems/Atom/RPI/Assets/ShaderLib/Atom/RPI/BitOperations.azsli

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+// Creates a bit mask for the first N bits
+// Example: GetBitMask(3) returns 00000000000000000000000000000111
+//                                              bitMaskLength ^
+uint GetBitMask(uint bitMaskLength)
+{
+    uint bitmask = (1u << bitMaskLength) - 1u;
+    return bitmask;
+}
+
+// Creates a bit for a range
+// startBitIncluded is the lowest bit set in the mask
+// endBitExcluded is the highest bit not set in the mask
+// Example: GetBitMask(2, 6) returns 00000000000000000000000000111100
+//                                             endBitExcluded ^   ^ startBitIncluded
+//
+uint GetBitMask(uint startBitIncluded, uint endBitExcluded)
+{
+    uint diff = endBitExcluded - startBitIncluded;
+    uint bitmask = GetBitMask(diff);
+    return (bitmask << startBitIncluded);
```

Comment:
It's unclear why this left shift would zero extend but the one in GetBitMask(uint) wouldn't?

Reply:
nm, I was ignoring the `-1` in `GetBitMask(uint)`